kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/TopicsController.java

```diff
@@ -126,87 +197,147 @@ public Mono<ResponseEntity<TopicsResponseDTO>> getTopics(String clusterName, @Va
-        .map(ResponseEntity::ok);
+
+    Mono<Void> validateAccess = accessControlService.validateAccess(AccessContext.builder()
+        .cluster(clusterName)
+        .clusterActions(ClusterAction.VIEW)
+        .topic(topicName)
+        .topicActions(VIEW, EDIT)
+        .build());
+
+    return validateAccess.then(
+        replicationFactorChange
+            .flatMap(rfc ->
+                topicsService.changeReplicationFactor(getCluster(clusterName), topicName, rfc))
+            .map(ResponseEntity::ok)
+    );
   }
 
   @Override
   public Mono<ResponseEntity<Void>> analyzeTopic(String clusterName, String topicName, ServerWebExchange exchange) {
-    return topicAnalysisService.analyze(getCluster(clusterName), topicName)
-        .thenReturn(ResponseEntity.ok().build());
+
+    Mono<Void> validateAccess = accessControlService.validateAccess(AccessContext.builder()
+        .cluster(clusterName)
+        .clusterActions(ClusterAction.VIEW)
+        .topic(topicName)
+        .topicActions(ANALYSIS_RUN)
+        .build());
+
+    return validateAccess.then(
+        topicAnalysisService.analyze(getCluster(clusterName), topicName)
+            .thenReturn(ResponseEntity.ok().build())
+    );
   }
 
   @Override
   public Mono<ResponseEntity<Void>> cancelTopicAnalysis(String clusterName, String topicName,
-                                                       ServerWebExchange exchange) {
+                                                        ServerWebExchange exchange) {
+    Mono<Void> validateAccess = accessControlService.validateAccess(AccessContext.builder()
+        .cluster(clusterName)
+        .clusterActions(ClusterAction.VIEW)
+        .topic(topicName)
+        .topicActions(ANALYSIS_RUN)
+        .build());
+
     topicAnalysisService.cancelAnalysis(getCluster(clusterName), topicName);
-    return Mono.just(ResponseEntity.ok().build());
+
+    return validateAccess.then(Mono.just(ResponseEntity.ok().build()));
   }
 
 
   @Override
   public Mono<ResponseEntity<TopicAnalysisDTO>> getTopicAnalysis(String clusterName,
                                                                  String topicName,
                                                                  ServerWebExchange exchange) {
-    return Mono.just(
+
+    Mono<Void> validateAccess = accessControlService.validateAccess(AccessContext.builder()
+        .cluster(clusterName)
+        .clusterActions(ClusterAction.VIEW)
+        .topic(topicName)
+        .topicActions(ANALYSIS_VIEW)
+        .build());
+
+    return validateAccess.then(Mono.just(
         topicAnalysisService.getTopicAnalysis(getCluster(clusterName), topicName)
             .map(ResponseEntity::ok)
-            .orElseGet(() -> ResponseEntity.notFound().build())
+            .orElseGet(() -> ResponseEntity.notFound().build()))
     );
   }
+
+  private Comparator<InternalTopic> getComparatorForTopic(
+      TopicColumnsToSortDTO orderBy) {
```

Comment:
I think it would be better to log what type of comparator have chosen for debug purposes

Reply:
I haven't touched a line here, just moved the methods, the diff is broken